### PR TITLE
Optimize union iterator

### LINKED ISSUE #1392: Maybe faster way to implement UI_ReadSorted
For now, redisearch will scan all active `IndexIterator` in `UnionIterator` for every call of `UI_ReadSorted()` to find the minimum Iterator, which is `O(n*m)` in time to read the whole `UnionIterator` where n is total number of item in `UnionIterator` and `m` is the number of `IndexIterator` in `UnionIterator`.

In my opinion, the better way to implement this function is to use a heap like `std::priority_queue` or write by hand to maintain the minimun `docId` which is `O(n*log(m))` in time to read the whole iterator.

When the number of `m` is too large, it is very slow to read. I've encountered this problem with a large `TagLexRange` query. One unique `tag` in result will lead to one `TrieMapNode` and then one `IndexIterator`. So the number of `IndexIterator` is large enough to effect the execute time.

### PR DESCRIPTION
**Pull request by ashtul**
fix #1392

### CODE REVIEW

**Review by MeirShpilraien — DISMISSED** [review]

**Review by rafie — DISMISSED** [review]

File: src/pytest/test.py [hunk 1]
```diff
@@ -124,48 +124,51 @@ def testUnion(env):
...
-            'nocontent', 'verbatim', 'limit', '0', '100')
-        env.assertEqual(101, len(res))
-        env.assertEqual(100, res[0])
+            res = r.execute_command(
+                'ft.search', 'idx', '(hello world)|((hello world)|(hallo world|werld) | hello world werld)',
+                'nocontent', 'verbatim', 'limit', '0', '100')
+            env.assertEqual(101, len(res))
+            env.assertEqual(100, res[0])
+        # test new Union Iterator
+        env.cmd('ft.config set UNION_ITERATOR_HEAP 1')
```
Comment (ashtul) [inline src/pytest/test.py:171]: testing new UI

File: src/pytest/test_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import unittest
+from includes import *
+from common import getConnectionByEnv, waitForIndex, sortedResults, toSortedFlatList
+from time import sleep, time
+from RLTest import Env
+import random
+
+
+def testNewUnionIteratorRead(env):
```
Comment (MeirShpilraien) [inline src/pytest/test_benchmark.py:11]: is this file is duplicate?

**Review by MeirShpilraien — COMMENTED** [review]
Do we have python test that goes through the new methods now that we set the threshold to 20? also it might be good idea to add another flow of tests that set it to zero to check all hour tests with the new flow..

**Comment by ashtul** [comment]
replaced by #1616

**Pull request closed without merging**